be/src/util/s3_util.cpp

```diff
@@ -73,6 +73,20 @@ bool to_int(std::string_view str, int& res) {
     return ec == std::errc {};
 }
 
+class CustomRetryStrategy final : public Aws::Client::DefaultRetryStrategy {
+public:
+    CustomRetryStrategy(int maxRetries) : DefaultRetryStrategy(maxRetries) {}
+
+    bool ShouldRetry(const Aws::Client::AWSError<Aws::Client::CoreErrors>& error,
+                     long attemptedRetries) const override {
+        if (attemptedRetries < m_maxRetries &&
+            error.GetResponseCode() == Aws::Http::HttpResponseCode::TOO_MANY_REQUESTS) {
+            return true;
```

Comment:
We should record something here if we are doing retry.
e.g. a retry counter or something useful for us to observe the status of requesting the object storage service.